### ISSUE TITLE
Study how we could add the possibility to execute multiple queries in one API call

### ISSUE THREAD

**Issue by ddolcimascolo** [linagora/jmap-client#34]
The idea is to minimize roundtrips to the backend.
The spec defines _back references_ that can help for the implementation I suppose...

**Comment by dbenchi** [comment]
If I got it well you mean doing stuffs like saveAsDraft and moveToOutbox in one roundtrip?

**Comment by ddolcimascolo** [comment]
Yes @abenchi. The [spec](http://jmap.io/spec.html#setfoos) defines back references so that you can reference an entity created earlier in _the same_ jmap operation. This will look like this:

```
[ 
  [ "setMessages", { "create": { "clientIdForCreatedMessage": { ... } } }, "#0" ] 
  [ "setMessages", { "update": { "clientIdForMoveMessage": { "id": "#clientIdForCreatedMessage", mailboxIds: [...] } } }, "#1" ] 
] 
```

This will make both operations run in a single HTTP request, which is great as far as performance is concerned.

**Comment by aduprat** [comment]
And the server is soo performent that you can send him hundred of commands at once.

**Comment by dbenchi** [comment]
great
